tests/components/airpatrol/conftest.py

```diff
@@ -0,0 +1,128 @@
+"""Common fixtures for the AirPatrol tests."""
+
+from unittest.mock import MagicMock, patch
+
+from airpatrol.api import AirPatrolAPI
+import pytest
+
+from homeassistant.components.airpatrol.const import DOMAIN
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from tests.conftest import MockConfigEntry
+
+
+@pytest.fixture
+def mock_api_response():
+    """Mock AirPatrol API response."""
+    with patch(
+        "airpatrol.api.AirPatrolAPI.get_data",
+    ) as mock_response:
+        yield mock_response
+
+
+@pytest.fixture
+def mock_api_set_climate_data():
+    """Mock AirPatrol API set temperature."""
+    with patch(
+        "airpatrol.api.AirPatrolAPI.set_unit_climate_data",
+    ) as mock_set_temp:
+        yield mock_set_temp
+
+
+@pytest.fixture
+def mock_api_authentication(mock_api):
+    """Mock AirPatrol API authentication success."""
+    with patch("airpatrol.api.AirPatrolAPI.authenticate", autospec=True) as mock_auth:
+        mock_auth.return_value = mock_api
+        yield mock_auth
+
+
+@pytest.fixture
+def mock_api():
+    """Mock AirPatrol API."""
+    api = MagicMock(spec=AirPatrolAPI)
+    api.get_unique_id.return_value = "test_user_id"
+    api.get_access_token.return_value = "test_access_token"
+    return api
```

Comment:
Instead of having all these separate fixtures, I personally prefer to mock out the full object.

```py
@pytest.fixture
def mock_airgradient_client() -> Generator[AsyncMock]:
    """Mock an AirGradient client."""
    with (
        patch(
            "homeassistant.components.airgradient.AirGradientClient",
            autospec=True,
        ) as mock_client,
        patch(
            "homeassistant.components.airgradient.config_flow.AirGradientClient",
            new=mock_client,
        ),
    ):
        client = mock_client.return_value
        client.get_latest_firmware_version.return_value = "3.1.4"
        yield client
```

This way you end up way more flexible, allowing you to add more functions with ease.